include/envoy/server/instance.h

```diff
@@ -188,6 +188,11 @@ class Instance {
    * @return information about the local environment the server is running in.
    */
   virtual const LocalInfo::LocalInfo& localInfo() PURE;
+
+  /**
+   * @return the server-wide cluster manager factory.
+   */
+  virtual Upstream::ClusterManagerFactory& clusterManagerFactory() PURE;
```

Comment:
It seems very strange to me that this has to be added to this interface. Why is this needed? (I will try to answer my own question as I review).